hotsos/core/ycheck/engine/properties/requires/common.py

```diff
@@ -78,6 +78,36 @@ def not_installed(self):
         return set(self.installed).symmetric_difference(_all)
 
 
+class PackageVersionOperations(object):
```

Comment:
Please add tests for this class.

Reply:
Yeah, I just want to get some feedback to know if this is the right thing.

Reply:
The reason you are adding this is because you want to compare versions for a binary that is not installed by a package and so therefore we cannot use the apt or snap helpers. This is very specific to Juju so imho we should implement the logic in the juju plugin itself. Having said that, since these versions are dpkg style versions we should try to leverage the existing logic rather than re-implementing it. Lets try to find a way to re-use the logic in the existing apt requires property to compare these versions even though they didn't come from an apt package. I can have a go at adding support for this in a separate patch to reduce complexity.

Reply:
yeah, I didn't do any version checking. Just used `distutils.version`. We can replace our `packaging.DPKGVersion` for `distutils.version.LooseVersion` which is more generic and I suppose more well-established.